airflow/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -0,0 +1,127 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from time import sleep
+from typing import Optional
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.redshift_data import RedshiftDataHook
+
+
+class RedshiftDataOperator(BaseOperator):
```

Comment:
Since the `RedshiftDataHook.cancel_statement()` method is being introduced, WDYT about adding the `on_kill()` method override for this operator which run the `cancel_statement()` method? This would give users an opportunity to kill the query from the UI if needed.

Reply:
I second @josh-fell here. Adding "on_kill" is often missing in the operators and that has some bad effects for our users.

Reply:
OK sounds good.  @vincbeck is working on the outstanding issues including this one.